File: C/ABC/ABC285B.c
#include <stdio.h>
#include <string.h>

int main(void)
{
    int N;
    char S[5001]; // 入力の最大長に応じてサイズを調整

    scanf("%d", &N);
    scanf("%s", S);

    for (int i = 0; i < N - 1; i++)
    {
        int count = 0;
        int broken = 0; // break したかどうか

        for (int j = i + 1; j < N; j++)
        {
            if (S[j - i - 1] != S[j])
            {
                count++;
            }
            else
            {
                printf("%d\n", count);
                broken = 1;
                break;
            }
        }
        if (!broken)
        {
            printf("%d\n", count);
        }
    }

    return 0;
}

File: python/ABC/ABC285B.py
N = int(input())
S = input()

for i in range(1, N):
    count = 0
    for j in range(5 - i + 1):
        if S[j] == S[j + i]:
            print(count)
            break
        count += 1
    else:
        print(count)
